### PR TITLE
Added latest version fetching functionality

### PR DESCRIPTION
**Pull request by achernevskii**
* If no version is provided for the connector, NewConnector init function will fetch WAPI schema, with all the supported versions, and use the latest version for connector's WAPI calls.

### CODE REVIEW

File: connector.go [hunk 1]
```diff
@@ -104,6 +107,63 @@ type Connector struct {
 	requestor      HttpRequestor
 }
 
+func getLatestVersion(hostCfg HostConfig, authCfg AuthConfig, transportCfg TransportConfig) (string, error) {
+	schemaUrl := fmt.Sprintf("https://%s/wapi/v1.0/?_schema", hostCfg.Host)
```
Comment (skudriavtsev) [inline connector.go:111]: We must take into account hostCfg.Port as well.

File: connector.go [hunk 2]
```diff
@@ -104,6 +107,63 @@ type Connector struct {
...
+	req, err := http.NewRequest(http.MethodGet, schemaUrl, nil)
+	if err != nil {
+		return "", err
+	}
+
+	req.SetBasicAuth(authCfg.Username, authCfg.Password)
```
Comment (skudriavtsev) [inline connector.go:117]: What will happen in the case when we do not use user's name and password at all? (certificate-based client's authentication)

File: connector.go [hunk 3]
```diff
@@ -104,6 +107,63 @@ type Connector struct {
...
+
+	tr := &http.Transport{
+		TLSClientConfig: &tls.Config{
+			InsecureSkipVerify: !transportCfg.SslVerify,
+		},
+	}
+
+	client := &http.Client{
+		Timeout:   time.Second * 10,
```
Comment (skudriavtsev) [inline connector.go:126]: We should not use hard-coded timeout values whenever possible. Please use some user-specified option (CLI option, environment variable, etc), either already existing or introduced in this PR.

File: connector.go [hunk 4]
```diff
@@ -104,6 +107,63 @@ type Connector struct {
...
+		Transport: tr,
+	}
+	resp, err := client.Do(req)
+	if err != nil {
+		return "", err
+	}
+
+	if resp.StatusCode != 200 {
```
Comment (skudriavtsev) [inline connector.go:134]: please use http.StatusOk instead

File: connector.go [hunk 5]
```diff
@@ -104,6 +107,63 @@ type Connector struct {
...
+		log.Printf("Non OK Response: %v", resp)
```
Comment (skudriavtsev) [inline connector.go:135]: This form of messages is appropriate for 'debug' level, but for other cases there should be more formal and more detailed message (ex. including the text from resp.Status, maybe  along with response's body content)

**Review by skudriavtsev — CHANGES_REQUESTED** [review]

File: connector.go (hunk 2 above)
Comment (achernevskii) [reply connector.go]: Updated the logic. Now client will use the requestor object, which uses client certs, and if username/password are configured, I will add them to the request.

File: connector.go (hunk 4 above)
Comment (achernevskii) [reply connector.go]: Updated the logic. Now I use the requestor object, which handles incorrect HTTP status codes.

File: connector.go (hunk 5 above)
Comment (achernevskii) [reply connector.go]: Updated the logic. Now I use the requestor object, which handles incorrect HTTP status codes.

**Pull request closed without merging**